africanus/calibration/utils.py

```diff
@@ -330,3 +406,41 @@ def _residual_vis_fn(time_bin_indices, time_bin_counts, antenna1,
                         array_type=":class:`numpy.ndarray`")
 except AttributeError:
     pass
+
+CORRUPT_VIS_DOCS = DocstringTemplate("""
+Corrupts model visibilities with arbitrary
+Jones terms.
+
+Parameters
+----------
+time_bin_indices : $(array_type)
+    The start indices of the time bins
+    of shape :code:`(utime)`
+time_bin_counts : $(array_type)
+    The counts of unique time in each
+    time bin of shape :code:`(utime)`
+antenna1 : $(array_type)
+    First antenna indices of shape :code:`(row,)`.
+antenna2 : $(array_type)
+    Second antenna indices of shape :code:`(row,)`
+jones : $(array_type)
+    Gains of shape :code:`(time, ant, chan, dir, corr)`
+    or :code:`(time, ant, chan, dir, corr, corr)`.
+model : $(array_type)
+    Model data values of shape :code:`(row, chan, dir, corr)`
+    or :code:`(row, chan, dir, corr, corr)`.
```

Comment:
``vis`` needs to be documented here. I would also put it before model in the argument list to be consistent with ``residual_vis``

```python
def corrupt_vis(time_bin_indices, time_bin_counts, antenna1,
                antenna2, jones, model, vis):
```